saleor/graphql/webhook/mutations.py

```diff
@@ -306,3 +316,80 @@ def perform_mutation(cls, _root, info: ResolveInfo, **data):
         manager = get_plugin_manager_promise(info.context).get()
         manager.event_delivery_retry(delivery)
         return EventDeliveryRetry(delivery=delivery)
+
+
+class WebhookDryRun(BaseMutation):
+    payload = JSONString(
+        description="JSON payload, that would be sent out to webhook's target URL."
+    )
+
+    class Arguments:
+        query = graphene.String(
+            description="The subscription query that defines the webhook event and its "
+            "payload.",
+            required=True,
+        )
+        object_id = graphene.ID(
+            description="The ID of an object to serialize.", required=True
+        )
+
+    class Meta:
+        description = "Performs a dry run of a webhook event."
+        permissions = (AppPermission.MANAGE_APPS,)
```

Comment:
Shouldn't be authenticated_staff_user, instead of `MANAGE_APPS`? To generate dry payload for example for Order, you need to have manage_orders and manage_apps. I think that it would be enough to require authenticated staff user, as the resources are protected by proper permissions on the resovlers

Reply:
You are right, thanks